epinorm/norm.py

```diff
@@ -320,27 +474,170 @@ def _add_missing_columns(self):
         self._data["longitude"] = None
         self._data["original_record_id"] = None
 
+    def _search_tokens_diff_order(self, areas, second):
+        """
+        This method uses the geocoder to make a query of areas (a list of tokens) and second (a string).
+        we vary the order of the tokens in areas as they could be wrong. We also remove them if we still don't have a result
+        it tried all permutations up to the point where it doesn't include tokens from areas anymore (only "second")
+        """
+
+        for l in range(len(areas), -1, -1):
+            for option in itertools.permutations(areas, l):
+
+                query = ", ".join(option) + ", " + second
+                locality = self._geocoder.get_feature(
+                    "search", {"query": query}, term=query, term_type="query"
+                )
+                if locality:
+                    return locality
+
+        return None
+
+        
+    def _find_full_locality(self, areas, countryName):
```

Comment:
Rename the argument "countryName" to "country_name" and update all references to it.